src/MessageHandler/VirusScanHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\MessageHandler;
+
+use App\Util\VirusScanUtil;
+use Doctrine\ORM\EntityManagerInterface;
+use Psr\Log\LoggerInterface;
+use Socket\Raw\Factory as SocketFactory;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Xenolope\Quahog\Client as QuahogClient;
+use App\Message\VirusScan;
+use App\Entity\File;
+use App\Repository\FileRepository;
+
+class VirusScanHandler implements MessageHandlerInterface
+{
+    /**
+     * The Entity Manager.
+     *
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * The File Repository.
+     *
+     * @var FileRepository
+     */
+    private $fileRepository;
+
+    /**
+     * The monolog logger.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * The virus-scanning utility holder.
+     *
+     * @var VirusScanUtil $scanner
+     */
+    private $scanner;
+
+
+    /**
+     * Constructor for this Controller, to set up default services.
+     *
+     * @param EntityManagerInterface $entityManager   The entity handler.
+     * @param FileRepository         $fileRepository  The file Repository.
+     * @param LoggerInterface        $virusScanLogger Name hinted virus_scan logger.
+     * @param VirusScanUtil          $scanner         A stream-based virus-scanning utility instance.
+    */
+    public function __construct(EntityManagerInterface $entityManager, FileRepository $fileRepository, LoggerInterface $virusScanLogger, VirusScanUtil $scanner)
+    {
+        $this->entityManager = $entityManager;
+        $this->fileRepository = $fileRepository;
+        $this->logger = $virusScanLogger;
+        $this->scanner = $scanner;
+    }
+
+    /**
+     * Invoke function to scan a file for viruses.
+     *
+     * @param VirusScan $virusScan The VirusScan message to be handled.
+     */
+    public function __invoke(VirusScan $virusScan)
+    {
+        $fileId = $virusScan->getFileId();
+        $file = $this->fileRepository->find($fileId);
```

Comment:
Once again, I think we should use use file path as a property in VirusScan, way less stuff to inject and things to look up, also less code. Plus extra bonus, we can use this scanner for anything that has a file path. Also, didn't we discuss that this scanner would be stream based? In that case the property would be a stream, and the virusscanner doesn't have to care about permissions.

Reply:
The utility is stream based, the message handler is not. Something has to translate a file to a stream.

Reply:
I can certainly load filepath into the message though.